src/components/checkbox/checkbox-group.spec.tsx

```diff
@@ -54,26 +51,31 @@ describe("CheckboxGroup", () => {
       ))}
     </CheckboxGroup>
   ));
-
+  type TestConfig = [string, string | boolean | number, string][];
   describe.each([
     ["legend", "foo"],
     ["required", true, "isRequired"],
     ["legendInline", true, "inline"],
     ["legendWidth", 30],
     ["legendAlign", "right"],
     ["legendSpacing", 2],
-  ])("when the %s prop is set", (propName, propValue, expectedPropName) => {
-    it("then it should be passed to the underlying Fieldset component", () => {
-      expect(
-        render({ [propName]: propValue })
-          .find("Fieldset")
-          .prop(expectedPropName || propName)
-      ).toBe(propValue);
-    });
-  });
+  ] as TestConfig)(
```

Comment:
What do you think about breaking this test fixture down into one or several smaller tests to verify whether the group legend element is rendered, with the correct styles, etc?
Given this approach is testing implementation details, I feel this would give us better confidence in these tests in the long run.

Reply:
I agree with you that there tests do look a bit confusing, but if you look at the resulting tests, everything looks clear what is being tested. As for the rendering of the legend and it's styles, we already have it covered by Chromatic snapshots.